drivers/entropy/entropy_nrf5.c

```diff
@@ -100,3 +230,49 @@ DEVICE_AND_API_INIT(entropy_nrf5, CONFIG_ENTROPY_NAME,
 		    entropy_nrf5_init, &entropy_nrf5_data, NULL,
 		    PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 		    &entropy_nrf5_api_funcs);
+
+static int entropy_nrf5_init(struct device *device)
+{
+	struct entropy_nrf5_dev_data *dev_data = DEV_DATA(device);
+
+	/* Locking semaphore initialized to 1 (unlocked) */
+	k_sem_init(&dev_data->sem_lock, 1, 1);
+	/* Synching semaphore */
+	k_sem_init(&dev_data->sem_sync, 0, 1);
+
+	init((struct rand *)dev_data->thr, RAND_THREAD_LEN,
+	     CONFIG_ENTROPY_NRF5_THR_THRESHOLD);
+	init((struct rand *)dev_data->isr, RAND_ISR_LEN,
+	     CONFIG_ENTROPY_NRF5_ISR_THRESHOLD);
+
+	/* Enable or disable bias correction */
+	if (IS_ENABLED(CONFIG_ENTROPY_NRF5_BIAS_CORRECTION)) {
+		NRF_RNG->CONFIG |= RNG_CONFIG_DERCEN_Msk;
+	} else {
+		NRF_RNG->CONFIG &= ~RNG_CONFIG_DERCEN_Msk;
+	}
+
+	NRF_RNG->EVENTS_VALRDY = 0;
+	NRF_RNG->INTENSET = RNG_INTENSET_VALRDY_Msk;
+
+	NRF_RNG->TASKS_START = 1;
+#if defined(CONFIG_BOARD_NRFXX_NWTSIM)
+	NRF_RNG_regw_sideeffects();
+#endif
+
+	IRQ_CONNECT(NRF5_IRQ_RNG_IRQn, CONFIG_ENTROPY_NRF5_PRI, isr_rand,
+		    DEVICE_GET(entropy_nrf5), 0);
+	irq_enable(NRF5_IRQ_RNG_IRQn);
+
+	return 0;
+}
+
+int entropy_get_entropy_thr(u8_t *buf, u16_t len)
```

Comment:
@cvinayak do we really need this one? In thread mode we could use the standard one really